ecommerce/extensions/payment/views/payu.py

```diff
@@ -55,31 +52,27 @@ def payment_processor(self):
     def dispatch(self, request, *args, **kwargs):
         return super(PayuPaymentExecutionView, self).dispatch(request, *args, **kwargs)
 
-    def _get_basket(self, payment_id):
+    def _get_basket(self, basket_id):
         """
-        Retrieve a basket using a payment ID.
+        Retrieve a basket using a basket ID.
 
         Arguments:
-            payment_id: payment_id received from PayPal.
+            basket_id: basket_id to look for.
 
         Returns:
-            It will return related basket or log exception and return None if
-            duplicate payment_id received or any other exception occurred.
-
+            It will return related basket or None if an error occurs
         """
+
+        if not basket_id:
+            return None
+
         try:
-            basket = PaymentProcessorResponse.objects.get(
-                processor_name=self.payment_processor.NAME,
-                transaction_id=payment_id
-            ).basket
+            basket_id = int(basket_id)
```

Comment:
The problem here is that the payment_id is not unique, whereas the basket_id is unique?

Reply:
Yes, but I think we should rewrite the payu processor since I have found bad practices in this code. This is just to fix the current error